aptos-move/move-examples/feature-sandbox/sources/sandbox_messaging.move

```diff
@@ -0,0 +1,213 @@
+    ///
+    /// You can see the generic phantom type for the MessageHolder.  This allows us to declare
+    /// a type that might not be used directly in the struct, but instead as an inner
+    /// generic type.  Drop and store allow the type to be dropped or stored in a holding type.
+    ///
+    /// Additionally, the MessageHolder has key, which allows it to be stored directly in global storage.
+    ///
+    /// These doc comments also work on structs as seen here
+    struct MessageHolder<phantom T: drop + store> has key {
+        /// The message being stored in the holder
+        ///
+        /// Strings in Aptos are only valid UTF-8
+        message: String,
+        /// An event handle for Aptos events associated with the message
+        ///
+        /// Here an event handle can be found with the specific type that is associated with the event below
+        ///
+        /// Generics can be set
+        message_change_events: event::EventHandle<T>,
+    }
+    //<:!:resource
+
+    /// An event for when the holding message changes
+    ///
+    /// This allows us to define a message that will be returned through the events API on the Aptos fullnode.
+    /// These messages are not accessible within move, and can only have one way communication out to the events
+    /// API.
+    struct MessageChangeEvent has drop, store {
+        /// The message that previously existed in the MessageHolder
+        old_message: String,
+        /// The message that is replacing the previous message in the MessageHolder
+        new_message: String,
+    }
+
+    // All annotations must go above doc comments
+    // This annotation denotes that the function is a view function.  This means that the function can be called from
+    // the view function API on the fullnode, and provides you a way to read data in Move through the API layer.
+    #[view]
+    /// Retrieves the message from the struct
+    ///
+    /// View functions allow you to return arbitrary data, so we can easily see the internal value of the MessageHolder
+    /// without doing extra parsing.
+    ///
+    /// This is also a public function.  This can be called by any other module or in any Move script.  Functions can
+    /// return any type.
+    ///
+    /// Functions can have doc comments, which will show up in documentation.
+    public fun get_message(address: address): String acquires MessageHolder {
+        // It is best practice to check if a type exists before reading it from global storage.
+        // This allows you to provide user friendly contextual error messages.
+        assert!(exists<MessageHolder<MessageChangeEvent>>(address), not_found(ENO_MESSAGE_HOLDER));
+
+        // This retrieves the value from storage to return it to another function.
+        // The *& clones the String so it doesn't return a reference
+        *&borrow_global<MessageHolder<MessageChangeEvent>>(address).message
```

Comment:
Just "borrow_global<MessageHolder<MessageChangeEvent>>(address).message" is good enough as String is copyable: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/move-stdlib/sources/string.move#L13

Reply:
I think we should take a moment to remove this old `*&` notation from the code base.  I suspect that it's being copied around / some old timers are doing it :)